dev/tools/controllerbuilder/pkg/codegen/mappergenerator.go

```diff
@@ -499,7 +509,12 @@ func (v *MapperGenerator) writeMapFunctionsForPair(out io.Writer, pair *typePair
 				}
 
 				oneof := protoField.ContainingOneof()
-				if oneof != nil {
+				if protoField.HasOptionalKeyword() {
```

Comment:
A lot of this complexity is because we are guessing the go field types.  It might easier to actually look at the target go type, and it might also save us some overrides!